src/service/client-report/client-report.service.ts

```diff
@@ -17,6 +17,10 @@ export class ClientReportService {
   ) {}
 
   async generateReport(client: string) {
+    const cidSharing = await this.getCidSharing(client);
+
+    if (!cidSharing.length) return null;
```

Comment:
I think that might not work as expected (which I assume is not generating reports for strings that aren't client addresses/ids?). CidSharing is supposed to be empty for all "honest" clients. 
I'd rely on `dataCapStatsService.fetchClientDetails` for this purpose instead.